### ISSUE TITLE
The commitlog check blocking tombstone GC in the repair mode

### ISSUE THREAD

**Issue by emdroid** [scylladb/scylladb#20686]
When working on the #15607, I've found there seems to be some kind of issue that is blocking the tombstone GC in the repair mode, in that the tombstones are not cleaned until the restart of the nodes.

This seems to be connected to the commitlog check that has been introduced in the #15060 - the check seems to be stuck to the time when the service has been started (i.e. seemingly to when the schema commitlog has been created, as the tables in question are schema tables using the schema commitlog).

However I reproduced the similar behavior also with the normal tables in the https://github.com/emdroid/scylladb/commit/14e72f1b86dd423a44f43ac61a35f0bae38c3f71 where the restart is also required for the tombstone GC to complete.

In particular, when checking the logs in the case of schema tables when performing the flush:
```
INFO  2024-09-12 17:14:21,654 [shard 0:strm] api - perform_keyspace_flush: keyspace=system_schema tables=[]
INFO  2024-09-12 17:14:21,654 [shard 0:strm] commitlog - Forcing new active segment
INFO  2024-09-12 17:14:21,654 [shard 0:strm] commitlog - Forcing new active segment: closing segment
DEBUG 2024-09-12 17:14:21,655 [shard 0:strm] commitlog - Not safe to delete segment CommitLog-4-3949080.log; dirty is b4dbb7b4-dc49-3fb5-b3bf-ce6e434832ca
```

So it seems that something is blocking the commitlog segment cleanup unless restarting the node.

The issue has also been mentioned in another test case in here: https://github.com/scylladb/scylladb/pull/19016/files#r1652322315

**Comment by emdroid** [comment]
The reproducer: https://github.com/emdroid/scylladb/commit/14e72f1b86dd423a44f43ac61a35f0bae38c3f71

**Comment by elcallio** [comment]
What table is b4dbb7b4-dc49-3fb5-b3bf-ce6e434832ca?
I assume that `CommitLog-4-3949080.log` is never released in your test case? 
It should be noted that the schema commitlog is _not_ forced a new segment in #15060 - the assumption being that it would have a lower general throughput, and thus not really warrant it. Maybe this hints that it too should be forced to flush into a new segment?

**Comment by emdroid** [comment]
Yeah, I'm not really sure how this exactly works, so you might have some ideas why this is being blocked...

**Comment by elcallio** [comment]
Slight overhead, some latency. Not much. Will test if it resolves this test issue.

**Comment by emdroid** [comment]
> Slight overhead, some latency. Not much. Will test if it resolves this test issue.

I wouldn't expect it to fix the https://github.com/emdroid/scylladb/commit/14e72f1b86dd423a44f43ac61a35f0bae38c3f71 as that is done for the regular tables. So there might be something else in play in fact (might not even be a commitlog based issue) - although the restarts there are needed to make the test pass.

But it should probably fix the test in #20394 in that it should not need the server restart afterwards.

**Comment by elcallio** [comment]
Ok, so first, I don't quite understand why the test only flushes first node? Afaict, there are at least two nodes in play, and flushing just one certainly does not guarantee the offending memables will be written etc...

If I just drop the restart and instead ensure to call flush on every node, the test seems to get the desired zero count fairly quickly...

Does not mean it might not be a good idea to add flushing to schema commitlogs, but does not seem to be needed here...

**Comment by emdroid** [comment]
Fixed by #20691.
